ENIGMAsystem/SHELL/Graphics_Systems/General/GStiles.cpp

```diff
@@ -15,19 +15,29 @@
 *** with this code. If not, see <http://www.gnu.org/licenses/>
 **/
 
-// Tile system
+
+#include "GSprimitives.h"
+#include "GSbackground.h"
+#include "GStextures.h"
+#include "GStiles.h"
+#include "GSmodel.h"
+#include "GStilestruct.h"
 #include "Universal_System/depth_draw.h"
-#include <algorithm>
-#include "../General/GSbackground.h"
 #include "Universal_System/background.h"
 #include "Universal_System/background_internal.h"
-#include "../General/GStextures.h"
-#include "GL3TextureStruct.h"
-#include "../General/GStiles.h"
-#include "../General/GLtilestruct.h"
-#include "../General/OpenGLHeaders.h"
-#include "../General/GSprimitives.h" //pr_trianglestrip
-#include "../General/GSmodel.h" //For batcher
+
+#include <algorithm>
+
+std::map<int,int> tile_layer_models;
+//Tile vector holds several values, like number of vertices to render, texture to use and so on
+//The structure is like this [render batch][batch info]
+//batch info - 0 = texture to use, 1 = vertices to render,
+std::map<int,std::vector<std::vector<int> > > tile_layer_metadata;
+
+// not allowed to include mathnc.h outside of SHELLmain
+namespace enigma_user {
+    bool point_in_rectangle(ma_scalar px, ma_scalar py, ma_scalar x1, ma_scalar y1, ma_scalar x2, ma_scalar y2);
```

Comment:
don't do this. move point_in_rectangle to a header seperate mathnc. Move ma_scalar to its own header seperate mathnc if you need to.

Reply:
1) Did Josh say that's ok? That's what I was originally going to do, but then, where do we draw the line at what user math functions can be used in the engine and which ones can't?
2) `ma_scalar` is actually in its own header, it's in `Universal_System/scalar.h` with the others:
https://github.com/enigma-dev/enigma-dev/blob/f574a63563d2523ba46a0e782f17c16edd0a76be/ENIGMAsystem/SHELL/Universal_System/scalar.h#L25

Reply:
collision_point is this same function and it's not hidden away. I think @JoshDreamland put the mathnc rule in place to compensate for bugs in JDI and to prevent us from doing things like using enigma's sin over std::sin

Reply:
That's what I had presumed, so not everything in `mathnc.h` really needs cock-blocked. I think I may hold off making a second header until a second pr, we'll see (at least now it's in 1 place instead of 4). Regardless, since `point_in_rectangle` is now in universal, `collision_point` could actually be made to use that to increase coverage. `collision_point` must be kept however because its logic is slightly more complicated.

Reply:
I definitely encourage putting this in a header. Don't we have other simple collisions functions to lump into that header? E.g, line-line, triangle-triangle, or polygon-polygon?

Reply:
Yes, we do, and I can separate them into another file I suppose.

Reply:
Addressed: 885b43dabe5fea07ca16df1db6499f13bb91dab8

I made an exception to just include `mathnc.h` for now... too lazy to split it up just for one function.